indico/modules/vc/models/vc_rooms.py

```diff
@@ -331,26 +331,38 @@ def get_linked_for_event(cls, event):
         """Get a dict mapping link objects to event vc rooms"""
         return {vcr.link_object: vcr for vcr in cls.find_for_event(event)}
 
-    def delete(self, user):
+    def delete(self, user, delete_all = 0):
         """Deletes a VC room from an event
 
         If the room is not used anywhere else, the room itself is also deleted.
 
         :param user: the user performing the deletion
+        :param delete_all: if set to 1 all events linked to a videoconference room will be deleted together with the room. Default set to false.
         """
-        Logger.get('modules.vc').info("Detaching VC room {} from event {} ({})".format(
-            self.vc_room, self.event_new, self.link_object)
-        )
-        db.session.delete(self)
-        db.session.flush()
-        if not self.vc_room.events:
-            Logger.get('modules.vc').info("Deleting VC room {}".format(self.vc_room))
+        if not delete_all:
```

Comment:
Why not do `if delete_all` and then detach it from all events - after that you can use the logic we already have instead of duplicating code.